File: jumbotravel-portal/src/api/domain/flight_product.js
class FlightProduct {

    constructor({
        product_id,
        product_code,
        name,
        description,
        brand,
        type,
        stock,
        max,
        saleprice
    }) {
        this.product_id = product_id;
        this.product_code = product_code;
        this.name = name;
        this.description = description;
        this.brand = brand;
        this.type = type;
        this.stock = stock;
        this.max = max;
        this.saleprice = saleprice;
    }

    getProductId() {
        return this.product_id;
    }

    getProductCode() {
        return this.product_code;
    }

    getName() {
        return this.name;
    }

    getDescription() {
        return this.description;
    }

    getBrand() {
        return this.brand;
    }

    getType() {
        return this.type;
    }

    getStock() {
        return this.stock;
    }

    getMax() {
        return this.max;
    }

    getSalePrice() {
        return this.saleprice;
    }

}

export default FlightProduct;

File: jumbotravel-api/domain/dto/flight_product.go
package dto

type FlightProduct struct {
	ProductID   *int     `json:"product_id"`
	ProductCode *int     `json:"product_code"`
	Name        *string  `json:"name"`
	Description *string  `json:"description"`
	Brand       *string  `json:"brand"`
	Type        *string  `json:"type"`
	Stock       *int     `json:"stock"`
	Max         *int     `json:"max"`
	SalePrice   *float64 `json:"sale_price"`
} // @name FlightProduct

func (v *FlightProduct) GetDestFields() []interface{} {
	return []interface{}{
		&v.ProductID,
		&v.ProductCode,
		&v.Name,
		&v.Description,
		&v.Brand,
		&v.Type,
		&v.Stock,
		&v.Max,
		&v.SalePrice,
	}
}

func (v *FlightProduct) New() {
	*v = FlightProduct{}
}

func (v *FlightProduct) Val() interface{} {
	return *v
}

type BookingProduct struct {
	FlightProduct
	Quantity *int `json:"quantity"`
}
